wfc3tools/embedsub.py

```diff
@@ -91,28 +91,43 @@ def embedsub(files):
         flt[1].header['sizaxis1'] = yaxis
         flt[1].header['sizaxis2'] = xaxis
 
-        for i in range(1, 4):
-            flt[i].header['crpix1'] = crpix1 + x1 - 1
-            flt[i].header['crpix2'] = crpix2 + y1 - 1
-            flt[i].header['ltv1'] = 0.0
-            flt[i].header['ltv2'] = 0.0
+        # loop through each possible fits extension and modify keywords if they exist
+        for i in range(1, len(flt)):
+            try:
+                _ = flt[i].header['crpix1'] # if crpix1 doesn't exist this will cause a KeyError
+                flt[i].header['crpix1'] = crpix1 + x1 - 1
+                flt[i].header['crpix2'] = crpix2 + y1 - 1
+                _ = flt[i].header['ltv1'] # if ltv1 doesn't exist this will cause a KeyError
+                flt[i].header['ltv1'] = 0.0
+                flt[i].header['ltv2'] = 0.0
+            except KeyError:
+                # if crpix1 doesn't exist pass to the next extension
+                pass
+                # Note: it's ok to keep this all under one try because there will 
+                # never be a scenario where crpix1 doesn't exist but ltv1 does.
+                # However, there are some extensions where crpix1 exists but ltv1 doesn't.
 
         # set the header value of SUBARRAY to False since it's now
         # regular size image
         flt[0].header['SUBARRAY'] = False
-
+        
         # Now write out the SCI, ERR, DQ extensions to the full-chip file
-        hdulist = fits.HDUList()
-        hdulist.append(fits.ImageHDU(flt[0].data, header=flt[0].header))
-        hdulist.append(fits.ImageHDU(sci, header=flt[1].header, name='SCI'))
-        hdulist.append(fits.ImageHDU(err, header=flt[2].header, name='ERR'))
-        hdulist.append(fits.ImageHDU(dq, header=flt[3].header, name='DQ'))
-
+        hdu_list = fits.open(filename)
```

Comment:
since you're now iterating above (in case there are more than 5 science extensions), i would do the same here.